src/js/stores/DCOSStore.js

```diff
@@ -129,12 +129,19 @@ class DCOSStore extends EventEmitter {
     let deploymentsList = MarathonStore.get('deployments');
     let serviceTree = MarathonStore.get('groups');
 
-    NotificationStore.addNotification(
-      'services-deployments',
-      'deployment-count',
-      deploymentsList.getItems().length
+    let deploymentListlength = deploymentsList.getItems().length;
+    let currentDeploymentCount = NotificationStore.getNotificationCount(
+      'services-deployments'
     );
 
+    if (deploymentListlength !== currentDeploymentCount) {
```

Comment:
we do this in order to not be constantly be emitting change events. this will prevent the complicated services logic to constantly re-render